### ISSUE TITLE
Objective JH-2: Expand “Open in QGIS on JupyterHub” work

### ISSUE THREAD

**Issue by batpad** [NASA-IMPACT/veda-jupyterhub#42]
## Motivation

We currently can launch QGIS (in a Linux desktop environment in your browser) on the JupyterHub, and via URL instruct it to open specific vector layers. This is the start of ‘Open in QGIS’ work, and should be expanded.

## Proposal

The expansion is in two ways:

 - Learn what kinds of datasets would be most important to open, and add methods to open those kinds of datasets
 - Work with the VEDA frontend team to figure out appropriate next steps for linking to specific datasets from the frontend to “Open in QGIS” in the hub

We will also perform additional maintenance and stabilization work on this integration to make sure users have a smooth experience throughout.

## Owners
 - @batpad (Development Seed)

## Success criteria

 - [ ] Add ability to open more types of datasets than standard vector formats from a public URL
 - [ ] Demo how a VEDA dataset can be opened in QGIS on the hub
 - [ ] In collaboration with VEDA UI team, make specific plan for integration into VEDA UI frontend

**Comment by batpad** [comment]
Spoke to @wildintellect a bit about this and what data / tile formats we should support.

For now, I think we want to support 3 types of datasets:

 - Vector data, eg. GeoJSON, fgb - this is currently supported already 
 - Raster data pointing to a GeoTIFF
 - XYZ tile layers

@sunu can you add some pointers of where in the code we should look to add methods to support different types, and then I guess this should be pretty easy to test.

**Comment by sunu** [comment]
@batpad The code that opens remote vector datasets as a layer is [here](https://github.com/sunu/jupyter-remote-qgis-proxy/blob/baf0d373c2f965a60bc6fe038bb04cacc8df8cf5/jupyter_remote_qgis_proxy/qgis/utils.py#L23). Right now, it should support any remote source that can be opened with `vsicurl`.

**Comment by batpad** [comment]
We have not done this yet.

Couple reasons:

 - It seemed hard to be able to work closely with the UI team this quarter to spec this out from a frontend perspective.
 - We discovered that we have some maintanence issues with the QGIS stac plugin that we need to work out. We did the work to update the plugin to use updated STAC URLs etc over in https://github.com/NASA-IMPACT/veda-jupyterhub/issues/52, but realize we need to think about long-term maintainability and feasibility of the STAC plugin. cc @brianmfreitag @yuvipanda as something to talk about.

It's still relatively trivial for us to have a working demo of opening an `xyz` Tile URL, being served by for eg. TiTiler, if we think that is useful to be able to show, but I think it's more useful to have a coherent plan along-with the UI team for what an integration looks like.
